Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -61,26 +66,21 @@ bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) con
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    batteryData.voltage = sections.at(1).toDouble(&ok);
 
-    batteryData.current = sections.at(2).toDouble();
+    batteryData.current = sections.at(2).toDouble(&ok);
```

Comment:
what happens if the voltage is wrong, and stores false in ok, then current is correct and changes ok to true? there will be no error even if the voltage is broken!